core/src/main/java/io/aiven/kafka/tieredstorage/metrics/MetricsRegistry.java

```diff
@@ -0,0 +1,268 @@
+    final MetricNameTemplate segmentDeleteBytesRateByTopicPartition;
+    static final String SEGMENT_DELETE_BYTES_TOTAL = SEGMENT_DELETE_BYTES + "-total";
+    final MetricNameTemplate segmentDeleteBytesTotal;
+    final MetricNameTemplate segmentDeleteBytesTotalByTopic;
+    final MetricNameTemplate segmentDeleteBytesTotalByTopicPartition;
+    static final String SEGMENT_DELETE_TIME = SEGMENT_DELETE + "-time";
+    static final String SEGMENT_DELETE_TIME_AVG = SEGMENT_DELETE_TIME + "-avg";
+    final MetricNameTemplate segmentDeleteTimeAvg;
+    final MetricNameTemplate segmentDeleteTimeAvgByTopic;
+    final MetricNameTemplate segmentDeleteTimeAvgByTopicPartition;
+    static final String SEGMENT_DELETE_TIME_MAX = SEGMENT_DELETE_TIME + "-max";
+    final MetricNameTemplate segmentDeleteTimeMax;
+    final MetricNameTemplate segmentDeleteTimeMaxByTopic;
+    final MetricNameTemplate segmentDeleteTimeMaxByTopicPartition;
+    static final String SEGMENT_DELETE_ERRORS = SEGMENT_DELETE + "-errors";
+    static final String SEGMENT_DELETE_ERRORS_RATE = SEGMENT_DELETE_ERRORS + "-rate";
+    final MetricNameTemplate segmentDeleteErrorsRate;
+    final MetricNameTemplate segmentDeleteErrorsRateByTopic;
+    final MetricNameTemplate segmentDeleteErrorsRateByTopicPartition;
+    static final String SEGMENT_DELETE_ERRORS_TOTAL = SEGMENT_DELETE_ERRORS + "-total";
+    final MetricNameTemplate segmentDeleteErrorsTotal;
+    final MetricNameTemplate segmentDeleteErrorsTotalByTopic;
+    final MetricNameTemplate segmentDeleteErrorsTotalByTopicPartition;
+
+    // Segment fetch metric names
+    static final String SEGMENT_FETCH = "segment-fetch";
+    static final String SEGMENT_FETCH_RATE = SEGMENT_FETCH + "-rate";
+    final MetricNameTemplate segmentFetchRequestsRate;
+    final MetricNameTemplate segmentFetchRequestsRateByTopic;
+    final MetricNameTemplate segmentFetchRequestsRateByTopicPartition;
+    static final String SEGMENT_FETCH_TOTAL = SEGMENT_FETCH + "-total";
+    final MetricNameTemplate segmentFetchRequestsTotal;
+    final MetricNameTemplate segmentFetchRequestsTotalByTopic;
+    final MetricNameTemplate segmentFetchRequestsTotalByTopicPartition;
+    static final String SEGMENT_FETCH_REQUESTED_BYTES = SEGMENT_FETCH + "-requested-bytes";
+    static final String SEGMENT_FETCH_REQUESTED_BYTES_RATE = SEGMENT_FETCH_REQUESTED_BYTES + "-rate";
+    final MetricNameTemplate segmentFetchRequestedBytesRate;
+    final MetricNameTemplate segmentFetchRequestedBytesRateByTopic;
+    final MetricNameTemplate segmentFetchRequestedBytesRateByTopicPartition;
+    static final String SEGMENT_FETCH_REQUESTED_BYTES_TOTAL = SEGMENT_FETCH_REQUESTED_BYTES + "-total";
+    final MetricNameTemplate segmentFetchRequestedBytesTotal;
+    final MetricNameTemplate segmentFetchRequestedBytesTotalByTopic;
+    final MetricNameTemplate segmentFetchRequestedBytesTotalByTopicPartition;
+
+    public MetricsRegistry() {
```

Comment:
Not sure if it's worth changing now, but all the fields seemingly can be initialized in place and the constructor is not needed